File: unidbg-api/src/main/java/com/vortexdbg/hook/ReplaceCallback.java
package com.vortexdbg.hook;

import com.vortexdbg.Emulator;
import com.vortexdbg.arm.HookStatus;

public abstract class ReplaceCallback {

    public  HookStatus onCall(Emulator<?> emulator, long originFunction) {
        return HookStatus.RET(emulator, originFunction);
    }

    public  HookStatus onCall(Emulator<?> emulator, HookContext context, long originFunction) {
        return onCall(emulator, originFunction);
    }

    public void postCall(Emulator<?> emulator, HookContext context) {
    }

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/hook/ReplaceCallback.kt
package com.vortexdbg.hook

import com.vortexdbg.Emulator
import com.vortexdbg.arm.HookStatus

abstract class ReplaceCallback {

    open fun onCall(emulator: Emulator<*>, originFunction: Long): HookStatus {
        return HookStatus.RET(emulator, originFunction)
    }

    open fun onCall(emulator: Emulator<*>, context: HookContext, originFunction: Long): HookStatus {
        return onCall(emulator, originFunction)
    }

    open fun postCall(emulator: Emulator<*>, context: HookContext) {
    }

}
